src/main/java/core/basesyntax/strategy/OperationStrategyImpl.java

```
@@ -0,0 +1,17 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.Map;
+
+public class OperationStrategyImpl implements OperationStrategy {
+    private final Map<FruitTransaction.Operation, OperationHandler> handlersMap;
+
+    public OperationStrategyImpl(Map<FruitTransaction.Operation, OperationHandler> handlersMap) {
+        this.handlersMap = handlersMap;
+    }
+
+    @Override
+    public OperationHandler get(FruitTransaction.Operation type) {
+        return handlersMap.get(type);
```

Consider handling cases where the requested operation type is not present in the `handlersMap`. You could return a default handler or throw an exception if the operation type is not found to prevent potential `NullPointerException`.